### PR TITLE
BR/MR

### PR DESCRIPTION
**Pull request by Anthogator**
Still not showing up in a campaign, but most of the rest should be good

### CODE REVIEW

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 1]
```diff
@@ -1,9 +1,5 @@
-; This file covers the Research, engineering and stats of all weapons
+[LW_WeaponPack.X2StrategyElement_LaserTechs]
 
-; ************************
-; *****RESEARCH COSTS*****
-; ************************
-[LWOTC_WeaponPack.X2StrategyElement_LaserTechs]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:6]: Why were these comments deleted?

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 2]
```diff
@@ -1,9 +1,5 @@
-; This file covers the Research, engineering and stats of all weapons
+[LW_WeaponPack.X2StrategyElement_LaserTechs]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:1]: Mod name is LWOTC_WeaponPack this won't add the ini files correctly. You'll need to correct this across the entire file.

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 3]
```diff
@@ -14,7 +10,6 @@ PULSE_LASER_WEAPONS_ALLOYCOST=0
 PULSE_LASER_WEAPONS_ELERIUMCOST=0
 PULSE_LASER_WEAPONS_RESEARCH_POINTS=2500
 
-[LWOTC_WeaponPack.X2StrategyElement_CoilTechs]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:17]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 4]
```diff
@@ -55,25 +50,22 @@ SniperRifle_LASER_SCHEMATIC_SUPPLYCOST=30
...
 
+MarksmanRifle_LASER_SCHEMATIC_SUPPLYCOST=30
+MarksmanRifle_LASER_SCHEMATIC_ALLOYCOST=5
+MarksmanRifle_LASER_SCHEMATIC_ELERIUMCOST=0
+
 Pistol_LASER_SCHEMATIC_SUPPLYCOST=10
 Pistol_LASER_SCHEMATIC_ALLOYCOST=2
 Pistol_LASER_SCHEMATIC_ELERIUMCOST=0
 
-[LWOTC_WeaponPack.X2Item_LWMagneticSchematics]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:62]: Why was this stuff deleted?

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 5]
```diff
@@ -90,71 +82,50 @@ SniperRifle_COIL_SCHEMATIC_SUPPLYCOST=125
...
 
+MarksmanRifle_COIL_SCHEMATIC_SUPPLYCOST=125
+MarksmanRifle_COIL_SCHEMATIC_ALLOYCOST=15
+MarksmanRifle_COIL_SCHEMATIC_ELERIUMCOST=5
+
 Pistol_COIL_SCHEMATIC_SUPPLYCOST=40
 Pistol_COIL_SCHEMATIC_ALLOYCOST=10
 Pistol_COIL_SCHEMATIC_ELERIUMCOST=5
 
-[LWOTC_WeaponPack.X2Item_LWBeamSchematics]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:97]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 6]
```diff
@@ -218,7 +195,6 @@ SHORT_LASER_RANGE[23] = -21
 SHORT_LASER_RANGE[24] = -25
 SHORT_LASER_RANGE[25] = -30
 
-; MIDSHORT: SMG
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:221]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 7]
```diff
@@ -183,14 +154,20 @@ SNIPERRIFLE_LASER_ISOUNDRANGE = 27
...
+MARKSMANRIFLE_LASER_UPGRADESLOTS = 2
+
 PISTOL_LASER_AIM = 5
 PISTOL_LASER_CRITCHANCE = 0
 PISTOL_LASER_ICLIPSIZE = 99
 PISTOL_LASER_ISOUNDRANGE = 15
 PISTOL_LASER_IENVIRONMENTDAMAGE = 5
 
 ; range profiles for laser
-; SHORTRANGE: SHOTGUN
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:193]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 8]
```diff
@@ -312,46 +318,22 @@ SNIPERRIFLE_LS_SUPPLYCOST=15
...
 
+MARKSMANRIFLE_LS_SUPPLYCOST=15
+MARKSMANRIFLE_LS_ALLOYCOST=1
+MARKSMANRIFLE_LS_ELERIUMCOST=0
+
 PISTOL_LS_SUPPLYCOST=5
 PISTOL_LS_ALLOYCOST=1
 PISTOL_LS_ELERIUMCOST=0
 
-[LWOTC_WeaponPack.X2Item_LWMagneticWeapons]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:319]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 9]
```diff
@@ -521,20 +548,128 @@ SNIPERRIFLE_CG_SUPPLYCOST=25
...
 PISTOL_CG_ELERIUMCOST=1
+ 
+[LW_WeaponPack.X2Ability_SMGAbilities]
+SMG_CONVENTIONAL_MOBILITY_BONUS = 3
+SMG_LASER_MOBILITY_BONUS=3
+SMG_MAGNETIC_MOBILITY_BONUS = 3
+SMG_COIL_MOBILITY_BONUS = 3
+SMG_BEAM_MOBILITY_BONUS = 3
 
-[LWOTC_WeaponPack.X2Item_LWBeamWeapons]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:528]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 10]
```diff
@@ -553,17 +688,118 @@ MIDSHORT_BEAM_RANGE[13] = -8
 MIDSHORT_BEAM_RANGE[14] = -12
 MIDSHORT_BEAM_RANGE[15] = -16
 MIDSHORT_BEAM_RANGE[16] = -20
-; TODO
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:556]: Please add back

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 11]
```diff
@@ -553,17 +688,118 @@ MIDSHORT_BEAM_RANGE[13] = -8
...
 
-[LWOTC_WeaponPack.X2Ability_SMGAbilities]
```
Comment (TNychka) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini:558]: Please add back

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_BRSchematics.uc [hunk 12]
```diff
@@ -0,0 +1,222 @@
+//---------------------------------------------------------------------------------------
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_BRSchematics.uc:1]: I know it's a pain, but can you put all of this in the con/las/mag/coil/beam schematics file?

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_BR_MR_Pack.uc [hunk 13]
```diff
@@ -0,0 +1,638 @@
+//---------------------------------------------------------------------------------------
+//  FILE:   X2Item_BR_MR_Pack.uc
+//---------------------------------------------------------------------------------------
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_BR_MR_Pack.uc:3]: can you put these in the con/las/mag.... weapons.uc files?

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 14]
```diff
@@ -80,19 +101,25 @@ var config int SNIPERRIFLE_CG_SUPPLYCOST;
...
+var config int MARKSMANRIFLE_CG_ELERIUMCOST;
+
 var config int PISTOL_CG_SUPPLYCOST;
 var config int PISTOL_CG_ALLOYCOST;
 var config int PISTOL_CG_ELERIUMCOST;
 
 static function array<X2DataTemplate> CreateTemplates()
 {
-	local array<X2DataTemplate> Weapons;
+    local array<X2DataTemplate> Weapons;
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:114]: whitespace

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 15]
```diff
@@ -139,17 +166,17 @@ static function X2DataTemplate CreateAssaultRifle_Coil_Template()
...
+	if (class'X2Item_LaserSchematics'.default.USE_SCHEMATICS)
 	{
 		Template.CreatorTemplateName = 'AssaultRifle_CG_Schematic'; // The schematic which creates this item
-		Template.BaseItem = 'AssaultRifle_MG'; // Which item this will be upgraded from
+		Template.BaseItem = 'AssaultRifle_MG'; // Which item this will be upgraded from		
 	}
 	else
 	{
-		Template.Requirements.RequiredTechs.AddItem(class'X2StrategyElement_CoilTechs'.default.CoilWeaponTech_Tier[0]);
+		Template.Requirements.RequiredTechs.AddItem(class'X2StrategyElement_LaserTechs'.default.CoilWeaponTech_Tier[0]);
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:179]: All of these should be coil techs not laser techs

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 16]
```diff
@@ -139,17 +166,17 @@ static function X2DataTemplate CreateAssaultRifle_Coil_Template()
...
-	Template.bInfiniteItem = class'X2Item_CoilSchematics'.default.USE_SCHEMATICS;
+	Template.CanBeBuilt = !class'X2Item_LaserSchematics'.default.USE_SCHEMATICS;
+	Template.bInfiniteItem = class'X2Item_LaserSchematics'.default.USE_SCHEMATICS;
 
-	if (class'X2Item_CoilSchematics'.default.USE_SCHEMATICS)
+	if (class'X2Item_LaserSchematics'.default.USE_SCHEMATICS)
 	{
 		Template.CreatorTemplateName = 'AssaultRifle_CG_Schematic'; // The schematic which creates this item
-		Template.BaseItem = 'AssaultRifle_MG'; // Which item this will be upgraded from
+		Template.BaseItem = 'AssaultRifle_MG'; // Which item this will be upgraded from
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:175]: whitespace

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 17]
```diff
@@ -214,17 +314,17 @@ static function X2DataTemplate CreateSMG_Coil_Template()
...
-	Template.bInfiniteItem = class'X2Item_CoilSchematics'.default.USE_SCHEMATICS;
+	Template.CanBeBuilt = !class'X2Item_LaserSchematics'.default.USE_SCHEMATICS;
+	Template.bInfiniteItem = class'X2Item_LaserSchematics'.default.USE_SCHEMATICS;
 
-	if (class'X2Item_CoilSchematics'.default.USE_SCHEMATICS)
+	if (class'X2Item_LaserSchematics'.default.USE_SCHEMATICS)
 	{
 		Template.CreatorTemplateName = 'SMG_CG_Schematic'; // The schematic which creates this item
-		Template.BaseItem = 'SMG_MG'; // Which item this will be upgraded from
+		Template.BaseItem = 'SMG_MG'; // Which item this will be upgraded from
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:323]: whitespace

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 18]
```diff
@@ -214,17 +314,17 @@ static function X2DataTemplate CreateSMG_Coil_Template()
...
+	if (class'X2Item_LaserSchematics'.default.USE_SCHEMATICS)
 	{
 		Template.CreatorTemplateName = 'SMG_CG_Schematic'; // The schematic which creates this item
-		Template.BaseItem = 'SMG_MG'; // Which item this will be upgraded from
+		Template.BaseItem = 'SMG_MG'; // Which item this will be upgraded from		
 	}
 	else
 	{
-		Template.Requirements.RequiredTechs.AddItem(class'X2StrategyElement_CoilTechs'.default.CoilWeaponTech_Tier[0]);
+		Template.Requirements.RequiredTechs.AddItem(class'X2StrategyElement_LaserTechs'.default.CoilWeaponTech_Tier[0]);
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:327]: should be coil techs

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 19]
```diff
@@ -479,7 +653,7 @@ static function X2DataTemplate CreatePistol_Coil_Template()
 	Template.WeaponPanelImage = "_Pistol";                       // used by the UI. Probably determines iconview of the weapon.
 
 	Template.ItemCat = 'weapon';
-	Template.WeaponCat = 'pistol';
+	Template.WeaponCat = 'pistol';
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:656]: whitespace

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 20]
```diff
@@ -506,23 +680,23 @@ static function X2DataTemplate CreatePistol_Coil_Template()
 	Template.Abilities.AddItem('Reload');
 
 	Template.SetAnimationNameForAbility('FanFire', 'FF_FireMultiShotBeamA'); // TODO : update with new animation if necessary
-
+
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:683]: whitespace

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 21]
```diff
@@ -497,7 +671,7 @@ static function X2DataTemplate CreatePistol_Coil_Template()
 
 	Template.OverwatchActionPoint = class'X2CharacterTemplateManager'.default.PistolOverwatchReserveActionPoint;
 	Template.InfiniteAmmo = true;
-
+
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:674]: whitespace

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc [hunk 22]
```diff
@@ -538,10 +712,11 @@ static function X2DataTemplate CreatePistol_Coil_Template()
 
 		Template.Requirements.RequiredEngineeringScore = 15;
 	}
-
+
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_CoilWeapons.uc:715]: whitespace and everything above needs to be coil

File: LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_LaserWeapons.uc [hunk 23]
```diff
@@ -6,17 +6,19 @@
...
+var config WeaponDamageValue BATTLERIFLE_LASER_BASEDAMAGE;
 var config WeaponDamageValue SMG_LASER_BASEDAMAGE;
 var config WeaponDamageValue LMG_LASER_BASEDAMAGE;
 var config WeaponDamageValue SHOTGUN_LASER_BASEDAMAGE;
 var config WeaponDamageValue SNIPERRIFLE_LASER_BASEDAMAGE;
+var config WeaponDamageValue MARKSMANRIFLE_LASER_BASEDAMAGE;
 var config WeaponDamageValue PISTOL_LASER_BASEDAMAGE;
 
-//�*****�Core properties and variables�for�weapons�*****
+// ***** Core properties and variables for weapons *****
```
Comment (TNychka) [inline LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_LaserWeapons.uc]: this file is ok except for the whitespace here and above

**Review by TNychka — CHANGES_REQUESTED** [review]

File: LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini [hunk 24]
```diff
@@ -90,71 +82,50 @@ SniperRifle_COIL_SCHEMATIC_SUPPLYCOST=125
...
-SMG_CONVENTIONAL_BASEDAMAGE=	(Damage=3, Spread=1, PlusOne=0, Crit=2, Pierce=0, Shred=0, Tag="", DamageType="Projectile_Conventional")
-SMG_CONVENTIONAL_AIM = 0
-SMG_CONVENTIONAL_CRITCHANCE = 0
-SMG_CONVENTIONAL_ICLIPSIZE = 3
-SMG_CONVENTIONAL_ISOUNDRANGE = 27
-SMG_CONVENTIONAL_IENVIRONMENTDAMAGE = 5
-SMG_CONVENTIONAL_UPGRADESLOTS = 1
-
-; **** WEAPON RANGE TABLES ****
-; MIDSHORT WEAPONS: SMG
```
Comment (Anthogator) [inline LWOTC_WeaponPack/Config/XComLW_WeaponPack.ini]: Will this table override the ini config values? or will the ini config values override this table?

**Pull request closed without merging**